File: app/src/main/java/io/mgba/UI/Activities/Interfaces/IIntroView.java
package io.mgba.UI.Activities.Interfaces;

import android.support.v7.app.AppCompatActivity;

import com.github.paolorotolo.appintro.AppIntroFragment;

import java.util.List;


public interface IIntroView {
    void addSlides(List<AppIntroFragment> slides);
    void savePreference(String key, String value);
    void savePreference(String key, boolean value);
    void showProgressDialog();
    void startActivity(Class<? extends AppCompatActivity> activity);
}

File: app/src/main/java/io/mgba/ui/activities/interfaces/IIntroView.kt
package io.mgba.ui.activities.interfaces

import com.github.paolorotolo.appintro.AppIntroFragment

import androidx.appcompat.app.AppCompatActivity

interface IIntroView {
    fun addSlides(slides: List<AppIntroFragment>)
    fun savePreference(key: String, value: String)
    fun savePreference(key: String, value: Boolean)
    fun showProgressDialog()
    fun startActivity(activity: Class<out AppCompatActivity>)
}
